site/docs/config-file.mdx

```diff
@@ -21,46 +21,81 @@ export const ExampleChart = () => {
+
+* `.cssnanorc.config.json` and `.cssnanorc` must contain a JSON object
+* `cssnano.config.js` must export the configuration as a JavaScript object
 
-We use configuration for `cssnano` using file name as `cssnano.config.js` , `cssnano` property in your `package.json`, using `cssnano.config.json` and `.cssnanorc` as well.
-We are using `cosmiconfig` in order to load cssnano config. Please read [here](https://github.com/davidtheclark/cosmiconfig) for more details
 
 ## Options
 
 ### `preset`
 
 - **Type:** `string` | `function` | `[string, Objects<preset options here>]` | `[function(preset options here)]`
 
-> For `string`, the name should be of type `cssnano-preset-<name>` and you need to use `name` alone eg : `preset : ['default', {}]`
+Pass a preset to choose a pre-configured set of optimizations. You can either import the preset package or use the preset name as a string.
 
-Contains lists of `postcss` plugins where each plugin does their own minification.
+With the preset as import:
 
-- **Example:**
-   
-  ```js
-  // cssnano.config.js
-  module.exports = {
-    preset: [require('cssnano-preset-default')]
-    
-    // or
-    preset: require('cssnano-preset-default')
-    
-    // or
-    preset: ['advanced', { discardComments: false }]
+```js
+cssnano({preset: require('cssnano-preset-default')})
+```
+
+Using a string is useful if you use a configuration file in the JSON format.
+When you use a string, if the preset is called `cssnano-preset-<name>`, use `name` alone:
+
+```js
+cssnano({preset: 'default'})
+```
+
+Preset themselves can take options.
+Pass options to the preset by using an array where the first element is the preset and the second is an object with the preset options. 
+You can specify a preset with the preset name as a string or by passing the result of importing the preset package.
+
+```js
+preset: [require('cssnano-preset-default'), {discardComments: false}]
+```
+
+
+You can also pass preset options when you use the preset name as a string:
+For example, here's how to deactivate the `discardComments` plugin when using the `advanced` preset:
+
+```js
+{preset: ['advanced', { discardComments: false }]}
+```
 
-    // or
-    preset: [require('cssnano-preset-default'), {discardComments: false}]
-  }
-  ```
 
 ### `plugins`
 
 - **Type:** `Array<'string' | 'function' | ['string' | 'function', Object<Options for the plugin here>]>`
 
-> If you want to pass config to the plugin, you need to use Array's of array for the plugins i.e `plugins: [ ['autoprefixer', {}] ]`
+In addition to the preset, you can pass a list of plugins to cssnano.
+This is equivalent to adding the plugins after cssnano in the PostCSS plugins array.
+If you want to configure the individual plugins, use an array of arrays:
 
-These plugins will run after once all presets operations are complete.
+```js
+plugins: [ ['autoprefixer', {}] ]
```

Comment:
Could it be clearer for a novice user to expand the example slightly?

```js
plugins: [ ['autoprefixer', { flexbox: false }], ... ]
```

or something like this to show multiple plugins configured individually..

```js
plugins: [ 
  ['autoprefixer', { flexbox: false }],
  ['pixrem', { rootValue: 12, replace: true }],
]
```

Reply:
Oh, scratch this comment. I see it's covered in the example below.